controllers/userController.js

```diff
@@ -22,13 +22,83 @@ exports.loginByWechat = function(req,res){
 	util.checkParam(req.body,['code'],function(err){
 		if (err) {
 			console.log(err);
-			res.send(util.wrapBody('Internal Error','E'));
+			res.send(util.wrapBody('Invalid Parameter','E'));
 		}else{
 			login(req,res,'wechat');
 		}
 	});	
 }
 
+exports.update = function(req,res){
+	// util.checkParam(req.body,['name','nickname'],function(err){
+	// 	if (err) {
+	// 		console.log(err);
+	// 		res.send(util.wrapBody('Invalid Parameter','E'));
+	// 		return;
+	// 	}
+	// });
+
+	var id = req.token.userId;
+
+	var user = {
+		name:req.body.name,
+		nickname:req.body.nickname,
+		skills:req.body.skills,
+		sector:req.body.sector
+	};
+
+	userRepository.updateById(id,user,function(err,result){
+		if (err) {
+			console.log(err);
+			res.send(util.wrapBody('Internal Error','E'));
+		}else{
+			res.send(util.wrapBody(result));
+		}
+	});
+}
+
+exports.getProfileById = function(req,res){
+
+	var id = req.params.id;
+
+	userRepository.findById(id,function(err,result){
+		if (err) {
+			console.log(err);
+			res.send(util.wrapBody('Internal Error','E'));
+		}else{
+			var responseBody = {
+				user:result
+			};
+
+			res.send(util.wrapBody(responseBody));
+		}
+	});
+}
+
+exports.listUser = function(req,res){
+	var conditions = {
```

Comment:
Also same as the above comments. Please ensure the behaviour is correct when some of the condition is missing in body.